### PR TITLE
Allow to switch off the license check for community distributions

### PR DESCRIPTION
**Pull request by artem-sidorenko**
lets have a similar approach as in chef/chef: https://github.com/chef/chef/pull/8354/files#diff-847fc1c7c0af04a65c570c6f4c3381e9R25

## Description

In order to be able to pass the `enforce_license` flag from the binstub, the EULA checking code had to be moved to some Thor-related method within early initialisation phase of application.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/base_cli.rb [hunk 1]
```diff
@@ -24,6 +25,32 @@ class << self
...
+      super(given_args, config)
+    end
+
+    # EULA acceptance
+    def self.check_license!
+      allowed_commands = ['-h', '--help', 'help', '-v', '--version', 'version']
+
+      require 'license_acceptance/acceptor'
+      begin
+        if (allowed_commands & ARGV.map(&:downcase)).empty? &&  # Did they use a non-exempt command?
```
Comment (zenspider) [inline lib/inspec/base_cli.rb]: Might just be my brain (and it often is!), but the inverse of this makes more sense to me. Something like:

```ruby
!(ARGV.map(&:downcase) - allowed_commands).empty? # man I wish English had a proper antonym for empty.
```

**Review by zenspider — COMMENTED** [review]
Just commentary

File: lib/inspec/base_cli.rb (hunk 1 above)
Comment (artem-sidorenko) [reply lib/inspec/base_cli.rb]: I did not to change the existing code, I just moved it without major changes. However I'm open for such changes/optimizations if needed

**Review by clintoncwolfe — APPROVED** [review]
Thanks - and this begins a refactor of cli.rb that I would love to continue :-)

File: lib/inspec/base_cli.rb (hunk 1 above)
Comment (zenspider) [reply lib/inspec/base_cli.rb]: Not needed. Just a suggestion. I'll get to it sooner or later.

**Review by zenspider — APPROVED** [review]

**Comment by miah** [comment]
I resolved the merge errors, lets get this merged today.

**Comment by miah** [comment]
We'll also need to resolve the chefstyle issues after merge; not a huge deal breaker.

**Review by miah — APPROVED** [review]

**Pull request merged**